src/wallet/gtest/test_wallet_zkeys.cpp

```diff
@@ -368,13 +364,11 @@ TEST(wallet_zkeys_tests, write_cryptedzkey_direct_to_db) {
     ASSERT_TRUE(wallet.EncryptWallet(strWalletPass));
     
     // adding a new key will fail as the wallet is locked
-    EXPECT_ANY_THROW(wallet.GenerateNewZKey());
+    EXPECT_ANY_THROW(wallet.GenerateNewSproutZKey());
     
     // unlock wallet and then add
     wallet.Unlock(strWalletPass);
-    auto address2 = wallet.GenerateNewZKey();
-    ASSERT_NE(boost::get<libzcash::SproutPaymentAddress>(&address2), nullptr);
-    auto paymentAddress2 = boost::get<libzcash::SproutPaymentAddress>(address2);
+    auto paymentAddress2 = wallet.GenerateNewSproutZKey();
```

Comment:
Shouldn't we still have the null assertion? The name change in this function does not ensure this check is done automatically for us.

Reply:
see below